### PR TITLE
Bump ugorji library and start using -tags codec.safe

### PR DESCRIPTION
**Pull request by mjudeikis**
### Which issue this PR addresses:

Fixes potential marshaling issues with codec

### What this PR does / why we need it:

WE had a bug where codec leaked data into other goroutine

### Test plan for issue

E2E

### Is there any documentation that needs to be updated for this PR?

no

Replaces https://github.com/Azure/ARO-RP/pull/1419

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -83,7 +83,7 @@ require (
 	github.com/rivo/uniseg v0.2.0 // indirect
 	github.com/sirupsen/logrus v1.7.1
 	github.com/syndtr/gocapability v0.0.0-20200815063812-42c35b437635 // indirect
-	github.com/ugorji/go/codec v1.2.5-0.20210320190651-a2bb12368408
+	github.com/ugorji/go/codec v1.2.5-0.20210326071331-11d01daad36c
```
Comment (troy0820) [inline go.mod:86]: Verified release fixes issue linked in comment 🎉

**Review by troy0820 — COMMENTED** [review]
LGTM

**Comment by mjudeikis** [comment]
There is something off with this PR. Somehow safe flag make create consistently fail in CI but not locally. 
Might be golang version related, marshaling or something else all together ;/

**Comment by mjudeikis** [comment]
Some strange flakes. needs debug

**Comment by mjudeikis** [comment]
@jewzaam why this is hold?

**Review by troy0820 — APPROVED** [review]
LGTM

**Pull request merged**